[Ember Migration] Debug Page

ajoykumardas12 opened the pull request:
Date: `29th Jan, 2024`

Developer Name: @ajoykumardas12 

---

## Issue Ticket Number:-
closes #738 
- #738 

## Description:

- Added debug page.
- Data is shown from API calls, not dummy data.
- Added Instagram id.
- Feature flags are hardcoded.
-  Apply privileges button is dummy, not storing any data in localStorage unlike vanilla version, because the functionality clearly would have to be changed again to make it useful.
- I have not followed every single style choices from last design, excluded the ones seemed unnecessary, please comment if anything should be different.
- The design is responsive at >300px.
- The font should be different I think, current one is using default font for www site, Raleway. The numbers are hard to read in this one.

Is Under Feature Flag

- [x] Yes
- [ ] No

Database changes

- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)

- [ ] Yes
- [x] No

Is Development Tested?

This PR is already big and I need little time exploring tests in ember related to api calls, so will make separate PR for tests. Issue ticket: #854 

### Add relevant Screenshot below ( e.g test coverage etc. )

Previous version (vanilla www site): 
Connected to local:
![screencapture-localhost-5500-debug-html-2024-01-24-11_26_42](https://github.com/Real-Dev-Squad/website-www/assets/87164140/53a25b04-20fa-4e6c-8085-326257600fda)

Connected to Prod:
![screencapture-dev-realdevsquad-debug-html-2024-01-24-10_57_08](https://github.com/Real-Dev-Squad/website-www/assets/87164140/fac63486-1715-4c03-afc0-cca77e95b54c)

New (Ember):

https://github.com/Real-Dev-Squad/website-www/assets/87164140/7fb3363b-c3d8-43ab-a570-8cb9c34b90c8

vinit717 commented on tests/integration/components/debug-grids-test.js:
test ?

ajoykumardas12 replied:
Will send different PR for tests.

ajoykumardas12 replied:
Issue #854

ajoykumardas12 replied:
Tests PR: #856

vinit717 commented on app/components/debug-grids.js:
```diff
@@ -0,0 +1,89 @@
+    }
+  }
+
+  checkAuth() {
+    this.store
+      .findRecord('user', 'self')
+      .then((user) => {
+        this.userData = user;
+        this.parseDebugData(this.userData);
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
```
You can use async await

ajoykumardas12 commented:
This wiki contains information related to development of debug page [Wiki Link](https://github.com/Real-Dev-Squad/website-www/wiki/%5BFeature%5D-%E2%80%90-Debug-Page)
cc: @satyam73

ajoykumardas12 commented on app/styles/debug.module.css:
```diff
@@ -0,0 +1,227 @@
+  color: var(--color-navyblue);
+}
+
+.debug-profile__name {
```
We are following BEM convention in www site.

ajoykumardas12 commented on app/components/debug-grids.js:
```diff
@@ -0,0 +1,89 @@
+    };
+  }
+
+  @action setPrivilages() {
+    this.toast.success(
+      'Your privileges are applied',
+      'Success!',
+      TOAST_OPTIONS,
+    );
+  }
```
Yes, this functionality is dummy currently. In vanilla version, the updated value was stored in localStorage, I didn't implement localStorage saving here because it wasn't serving any purpose.

prakashchoudhary07 replied:
Then do we need that function?

ajoykumardas12 replied:
Do you mean keeping the dummy button only without toast?

prakashchoudhary07 replied:
Yes both, if it isn't doing anything, should we keep it?

ajoykumardas12 replied:
True, it serves no purpose currently, so removing the functionality.

satyam73 commented on app/components/debug-grids.hbs:
can we use BEM convention for css classes?

ajoykumardas12 replied:
I tried to follow BEM convention. Can you please point out what I did wrong?

satyam73 commented on app/components/debug-grids.js:
```diff
@@ -0,0 +1,75 @@
+    }
+  }
+
+  async authenticateUserAndFetchdata() {
+    try {
+      const user = await this.store.findRecord('user', 'self');
+      this.userData = user;
+      this.parseDebugData(this.userData);
+    } catch (error) {
+      console.log('Failed to authenticate');
+    } finally {
+      this.isLoading = false;
+    }
+  }
```
this api is already present in login service, let's use from that one

satyam73 commented on app/styles/debug.module.css:
```diff
@@ -0,0 +1,235 @@
+}
+
+.debug-user__roles-container ul {
+  margin-top: 0.08rem;
```
for very small dimensions we can use px i.e for <=10px

satyam73 commented on app/styles/debug.module.css:
```diff
@@ -0,0 +1,235 @@
+  margin-bottom: 1rem;
+}
+
+@media only screen and (width > 1600px) {
```
are we using this breakpoint?

ajoykumardas12 replied:
No, this big breakpoint isn't used. I used this to keep the design well for large screens too 
![Screenshot_15](https://github.com/Real-Dev-Squad/website-www/assets/87164140/d2a94745-168b-4ad4-9393-7ae9c998e7d6)

ajoykumardas12 replied:
Updated

satyam73 commented on app/styles/debug.module.css:
```diff
@@ -0,0 +1,235 @@
+  }
+}
+
+@media only screen and (width <= 940px) {
```
are we using this breakpoint in our code?

ajoykumardas12 replied:
Changed to 922px, btw 922px was used only once before this.

satyam73 commented on app/components/debug-grids.hbs:
```diff
@@ -0,0 +1,83 @@
+      </div>
+      <h2 class="debug-profile__name">{{this.debugProfileData.fullName}}</h2>
+    </div>
+    <div class="debug-social debug__grid-item">
```
```suggestion
    <div class="debug__social debug__grid-item">
```

ajoykumardas12 replied:
Can `debug-social` not be a block?

satyam73 commented on app/components/debug-grids.hbs:
```diff
@@ -0,0 +1,83 @@
+        {{/each-in}}
+      </ul>
+    </div>
+    <div class="debug-user debug__grid-item">
```
```suggestion
    <div class="debug__user debug__grid-item">
```

ajoykumardas12 replied:
Why `debug-user` shouldn't be a block?

ajoykumardas12 replied:
Updated

satyam73 commented on app/components/debug-grids.hbs:
```diff
@@ -0,0 +1,83 @@
+              {{value}}</p>
+          </li>
+        {{/each-in}}
+        <li class="debug-user__roles-container">
```
```suggestion
        <li class="debug__user-roles">
```

ajoykumardas12 replied:
Updated

ajoykumardas12 commented:
401 image was added already in unauthorized component, but that component doesn't go well for this situation as debug page isn't under any authorization rather is under authentication. So, I have added a new component.

Updated visuals (unauthenticated): 

https://github.com/Real-Dev-Squad/website-www/assets/87164140/51c57a5d-fc90-4baa-b205-7b31cab1e7c8

satyam73 reviewed (approved):
LGTM
